Introduce the API documentation for risk-providers and risk-events APIs (#1827)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/risk-events/index.md

---
title: Risk Events
category: management
---

# Risk Events API

> The Okta RiskEvents API provides the ability for RiskProviders to send RiskEvents to Okta.

Reviewer: ```suggestion
The Okta Risk Events API provides the ability for third-party Risk Providers to send Risk Events to Okta.
```

Reviewer: fyi, as per style, updated to make separate words. Also added in "third-party", which I believe is correct, but highlighting here.

Regarding terminology: should we standardize on using "risk signals" or "risk events"? I generally see both used interchangeably.

## Get Started
Explore the Risk Events API: [![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/1c449b51a4a0adf90198)

## Risk Events Operations
The Risk Events API has the following operations:

* [Send RiskEvents](#send-riskevents)

### Send Risk Events

<ApiOperation method="post" url="/api/v1/risk/events/ip" />

> A RiskProvider can send RiskEvents to Okta using this API.

Reviewer: ```suggestion
A Risk Provider can send Risk Events to Okta using this API.
```

#### Request body

> The request body should include an array of [RiskEvents](#riskevent-object). A maximum of 20 events can be included in a request. Each RiskEvent can include the following properties:

Reviewer: ```suggestion
The request body should include an array of [Risk Events](#riskevent-object). A maximum of 20 events can be included in a request. Each Risk Event can include the following properties:
```

| Property    | Type           | Description   |
| ----------- | -------------- | ------------- |
| `timestamp` | String | Time stamp at which the the event is produced (Expressed as a UTC time zone using ISO 8601 format: yyyy-MM-dd'T'HH:mm:ss.SSS'Z'). This is a required field. |
| `expiresAt` | String | Time stamp at which the event expires (Expressed as a UTC time zone using ISO 8601 format: yyyy-MM-dd'T'HH:mm:ss.SSS'Z'). If this optional field is not included, Okta automatically expires the event 24 hours after the `timestamp`. |
| `subjects` | List | List of [Risk Subjects](#risksubject-object). A max of 50 subjects can be included in an event |

#### Response body

Http 202 (Accepted) is returned if the request is successful. If there are validation errors, the API returns a 400.

#### Request example

> This request sends RiskEvents to Okta

Reviewer: ```suggestion
This request sends Risk Events to Okta
```

```bash
curl -X POST \
-H "Accept: application/json" \
-H "Content-Type: application/json" \
-H "Authorization: Bearer ${access_token}" \
-d '{
    [
      {
       "timestamp": "2021-01-20T00:00:00.001Z",
       "subjects": [
          {
            "ip": "6.7.6.7",
            "riskLevel": "MEDIUM"
          },
          {
            "ip": "1.1.1.1",
            "riskLevel": "HIGH" ,
            "message": "Detected Attack tooling and suspicious activity"
          }
        ]
      },
      {
       "timestamp": "2021-01-20T01:00:00.001Z",
       "subjects": [
          {
            "ip": "6.7.6.7",
            "riskLevel": "LOW"
          },
          {
            "ip": "2.2.2.2",
            "riskLevel": "HIGH"
          }
        ]
      }
    ]
}' "https://${yourOktaDomain}/api/v1/risk/events/ip"
```

#### Response example
```http
HTTP/1.1 202 Accepted
```

> ## RiskEvents objects

Reviewer: ```suggestion
## Risk Events objects
```

> ### RiskEvent object

Reviewer: ```suggestion
### Risk Event object
```

The RiskEvent object has the following properties:

| Property    | Type           | Description   |
| ----------- | -------------- | ------------- |
| `timestamp` | String | Time stamp at which the event is produced (Expressed as a UTC time zone using ISO 8601 format: yyyy-MM-dd'T'HH:mm:ss.SSS'Z'). This is a required field. |
| `expiresAt` | String | Time stamp at which the event expires (Expressed as a UTC time zone using ISO 8601 format: yyyy-MM-dd'T'HH:mm:ss.SSS'Z'). If this optional field is not included, Okta automatically expires the event 24 hours after the `timestamp`. |
| `subjects` | List | List of [Risk Subjects](#risksubject-object) |

> #### RiskEvent example

Reviewer: ```suggestion
#### Risk Event example
```

```json
{
       "timestamp": "2021-01-20T00:00:00.001Z",
       "subjects": [
          {
            "ip": "6.7.6.7",
            "riskLevel": "MEDIUM"
          },
          {
            "ip": "1.1.1.1",
            "riskLevel": "HIGH" ,
            "message": "Detected Attack tooling and suspicious activity"
          }
        ]
}
```

> ### RiskSubject object

Reviewer: ```suggestion
### Risk Subject object
```

> The RiskSubject object has the following properties:

Reviewer: ```suggestion
The Risk Subject object has the following properties:
```

---
title: RiskEvents
category: management
---

> # RiskEvents API

Author: I dont have a strong preference here and can change it to `Risk Events`

> The Okta RiskEvents API provides the ability for RiskProviders to send RiskEvents to Okta.

Author: Ack. Once we have the link to the guide, I can refer that here.
